packages/context/src/resolution-session.ts

```diff
@@ -338,6 +338,13 @@ export interface ResolutionOptions {
    * will return `undefined` instead of throwing an error.
    */
   optional?: boolean;
+
+  /**
+   * A boolean flag to control if a proxy should be created to apply
+   * interceptors for the resolved value. It's only honored for bindings backed
+   * by a class.
```

Comment:
We should let the users know when this flag was ignored because the binding was not backed by a class.

1. You can print a `debug` statement, this way we don't pollute console with excessive output. 
2. Alternatively, I am also fine with throwing an error to let the programmer know about the error they made in their code.

I slightly prefer the second option (throw an error).

Reply:
> We should let the users know when this flag was ignored because the binding was not backed by a class.
>
> 1. You can print a `debug` statement, this way we don't pollute console with excessive output. 
> 2. Alternatively, I am also fine with throwing an error to let the programmer know about the error they made in their code.
>
> I slightly prefer the second option (throw an error).

☝️